implementations/rust/ockam/ockam_api/src/nodes/service.rs

```diff
@@ -379,119 +377,69 @@ impl NodeManager {
     }
 
     /// Resolve project ID (if any), create secure channel (if needed) and create a tcp connection
-    ///
-    /// Returns the secure channel worker address (if any) and the remainder
-    /// of the address argument.
+    /// Returns [`ConnectionInstance`]
     pub(crate) async fn connect(
```

Comment:
This method was completely rewritten to further generalize the `MultiAddr` handling